cicd/2-cicd/cicd.template.yml

```diff
@@ -26,11 +26,19 @@ Parameters:
     Type: String
     Description: The Resource Id component of the CodeStar connection ARN for the code-dot-org GitHub repository
     Default: 7df08bcf-9883-42e8-8f5e-d083419d4fe9
+  Mode:
+    Type: String
+    Description: Whether this should create a pipeline for an 'adhoc' deployment or 'standard' pipeline.
+    Default: standard
+    AllowedValues: [adhoc, standard]
 
 Conditions:
-  TargetsMainBranch: !Equals 
-    - !Ref GitHubBranch
-    - main
+  TargetsMainBranch: !Equals [ !Ref GitHubBranch, main ]
+  DeployToAdhoc: !Equals [!Ref Mode, adhoc ]
+  # DeployToTest and DeployToProd are the same right now
+  DeployToTest: !Equals [!Ref Mode, standard ]
+  DeployToProd: !Equals [ !Ref Mode, standard ]
```

Comment:
To clarify - what are these conditions for? Wouldn't the pipeline always deploy to both test and prod?

Reply:
Yeah, I went down a route of a scenario where you might want to deploy to Test but not Prod, but didn't see the value.

I could use two conditions [AdhocMode, StandardMode]. My dilemma is whether to name the condition based on the inputs, or based on the output the condition enables.

Reply:
cleaned these up.